Us110 raw data

rdsaliba opened the pull request:
Create a screen based on the mock-up. 
 The screen is accessible from the "System information" screen.

rdsaliba committed:
Adjusting UI for raw data implementation

Created Tab pane into the System Information screen. User can now select the Raw Data tab to diplay the table value items.

rdsaliba committed:
Update SystemInfoController.java

The controller is ready to receive logic to display the asset's raw data

PaulMicu commented on JavaFXUserInterface/src/main/java/SystemInfoController.java:
```diff
@@ -31,6 +31,37 @@
 import java.util.ResourceBundle;
 
 public class SystemInfoController implements Initializable {
+    private static final String S1_COL = "s1";
+    private static final String S2_COL = "s2";
+    private static final String S3_COL = "s3";
+    private static final String S4_COL = "s4";
+    private static final String S5_COL = "s5";
+    private static final String S6_COL = "s6";
+    private static final String S7_COL = "s7";
+    private static final String S8_COL = "s8";
+    private static final String S9_COL = "s9";
+    private static final String S10_COL = "s10";
+    private static final String S11_COL = "s11";
+    private static final String S12_COL = "s12";
+    private static final String S13_COL = "s13";
+    private static final String S14_COL = "s14";
+    private static final String S15_COL = "s15";
+    private static final String S16_COL = "s16";
+    private static final String S17_COL = "s17";
+    private static final String S18_COL = "s18";
+    private static final String S19_COL = "s19";
+    private static final String S20_COL = "s20";
+    private static final String S21_COL = "s21";
+    private static final String S22_COL = "s22";
+    private static final String S23_COL = "s23";
+    private static final String S24_COL = "s24";
+    private static final String S25_COL = "s25";
+    private static final String S26_COL = "s26";
+    private final String CYCLE_COL = "Cycle";
+    private final String OP1_COL = "OP1";
+    private final String OP2_COL = "OP2";
+    private final String OP3_COL = "OP3";
```
This is not dynamic enough, the amount and names of the attributes have to come from the database, not hard coded

PaulMicu commented on JavaFXUserInterface/src/main/java/SystemInfoController.java:
```diff
@@ -224,4 +266,61 @@ void deleteDialog(MouseEvent mouseEvent) {
+            return row;
+        });
+
+        TableColumn cycle = new TableColumn(CYCLE_COL);
+
+        TableColumn op1 = new TableColumn(OP1_COL);
+        TableColumn op2 = new TableColumn(OP2_COL);
+        TableColumn op3 = new TableColumn(OP3_COL);
+
+        TableColumn s1 = new TableColumn(S1_COL);
+        TableColumn s2 = new TableColumn(S2_COL);
+        TableColumn s3 = new TableColumn(S3_COL);
+        TableColumn s4 = new TableColumn(S4_COL);
+        TableColumn s5 = new TableColumn(S5_COL);
+        TableColumn s6 = new TableColumn(S6_COL);
+        TableColumn s7 = new TableColumn(S7_COL);
+        TableColumn s8 = new TableColumn(S8_COL);
+        TableColumn s9 = new TableColumn(S9_COL);
+        TableColumn s10 = new TableColumn(S10_COL);
+        TableColumn s11 = new TableColumn(S11_COL);
+        TableColumn s12 = new TableColumn(S12_COL);
+        TableColumn s13 = new TableColumn(S13_COL);
+        TableColumn s14 = new TableColumn(S14_COL);
+        TableColumn s15 = new TableColumn(S15_COL);
+        TableColumn s16 = new TableColumn(S16_COL);
+        TableColumn s17 = new TableColumn(S17_COL);
+        TableColumn s18 = new TableColumn(S18_COL);
+        TableColumn s19 = new TableColumn(S19_COL);
+        TableColumn s20 = new TableColumn(S20_COL);
+        TableColumn s21 = new TableColumn(S21_COL);
+        TableColumn s22 = new TableColumn(S22_COL);
+        TableColumn s23 = new TableColumn(S23_COL);
+        TableColumn s24 = new TableColumn(S24_COL);
+        TableColumn s25 = new TableColumn(S25_COL);
+        TableColumn s26 = new TableColumn(S26_COL);
+
+
+        table.setId("RawDataTable");
+        table.getColumns().addAll(cycle, op1, op2, op3, s1,s2,s3,s4,s5,s6,s7,s8,s9,s10,
+                s11, s12, s13, s14, s15, s16, s17, s18, s19, s20, s21, s22, s23, s24, s25, s26);
```
same as my other comment, this is not flexible enough, don't hard code the sensors directly into the code but dynamically generate them from the info stored in the db

jeffbuonamici replied:
You need to loop over the sensor data, and create the table columns dynamically.

PaulMicu reviewed (changes requested):
Please see comments

jeffbuonamici commented on JavaFXUserInterface/src/main/java/SystemInfoController.java:
```diff
@@ -224,4 +266,61 @@ void deleteDialog(MouseEvent mouseEvent) {
+    public void generateRawDataTable() {
+        TableView table = new TableView();
+
+        // When TableRow is clicked, send data to SystemInfo scene.
+        table.setRowFactory(tv -> {
+            TableRow<Asset> row = new TableRow<>();
+            row.setOnMouseClicked(event -> {
+                uiUtilities.changeScene(event, row, "/SystemInfo", row.getItem());
+            });
+            return row;
+        });
```
This code is unnecessary.

jeffbuonamici committed:
#187 Fixed a resizing issue.

jeffbuonamici committed:
#187 Displays RAW data for an Asset.

rdsaliba committed:
Renamed variable for readability

rdsaliba committed:
Correcting minor display variation

systemType id was already used in another view. It was inheriting its font value which made it disproportionate to the rest of the labels

PaulMicu commented on JavaFXUserInterface/src/main/java/SystemInfoController.java:
```diff
@@ -27,10 +34,42 @@
 import java.util.ResourceBundle;
 
 public class SystemInfoController implements Initializable {
+    private static final String S1_COL = "s1";
+    private static final String S2_COL = "s2";
+    private static final String S3_COL = "s3";
+    private static final String S4_COL = "s4";
+    private static final String S5_COL = "s5";
+    private static final String S6_COL = "s6";
+    private static final String S7_COL = "s7";
+    private static final String S8_COL = "s8";
+    private static final String S9_COL = "s9";
+    private static final String S10_COL = "s10";
+    private static final String S11_COL = "s11";
+    private static final String S12_COL = "s12";
+    private static final String S13_COL = "s13";
+    private static final String S14_COL = "s14";
+    private static final String S15_COL = "s15";
+    private static final String S16_COL = "s16";
+    private static final String S17_COL = "s17";
+    private static final String S18_COL = "s18";
+    private static final String S19_COL = "s19";
+    private static final String S20_COL = "s20";
+    private static final String S21_COL = "s21";
+    private static final String S22_COL = "s22";
+    private static final String S23_COL = "s23";
+    private static final String S24_COL = "s24";
+    private static final String S25_COL = "s25";
+    private static final String S26_COL = "s26";
+    private final String CYCLE_COL = "Cycle";
+    private final String OP1_COL = "OP1";
+    private final String OP2_COL = "OP2";
+    private final String OP3_COL = "OP3";
```
This code is never used and must be removed

PaulMicu reviewed (changes requested):
Some code needs to be removed(see comment) and the table is missing the Cycles column so that needs to be added

rdsaliba committed:
Updating TableView to display Cycle as first column

rdsaliba committed:
Merge branch 'develop' into US110_RawData

rdsaliba commented:
Please see attached image of the system with the recent changes.
<img width="1172" alt="Screen Shot 2021-01-21 at 12 20 02 AM" src="https://user-images.githubusercontent.com/17152773/105283471-8af12b00-5b7e-11eb-8a19-e93ceb4afb14.png">

rdsaliba commented:
The data list is not currently being updated when new measures are coming in. Needs to be fixed yet.

JayChouteau commented:
@rdsaliba Is the pull request only for issue  #120 or for both #120 and #88 ?? I thought this was for 88 only. This changes the scope of things.

PaulMicu reviewed (approved):
**accidentally approved but meant to request changes** the order should be reversed, it makes alot more sense that the info at the top is the most recent

PaulMicu reviewed (changes requested):
the order should be reversed, it makes alot more sense that the info at the top is the most recent

PaulMicu commented:
> @rdsaliba Is the pull request only for issue #120 or for both #120 and #88 ?? I thought this was for 88 only. This changes the scope of things.

I think this is correct, the issue is just to display the latest raw data, the other issue is for making sure the view updates with new measurements

rdsaliba commented:
@PaulMicu User can select the cycle column to sort by asc/dsc in order to see latest data.
As per the acceptance criteria of the Issue, it is not required to display by latest first. Tho, I agree it could be implemented this way in a future enhancement to the system. Then, maybe also we could add another column for "date"...
Yes this is just #120, I got confused.

PaulMicu commented:
@rdsaliba  I disagree, just because it's not spelled out word for word in the acceptance criteria does not mean it should not be done. But if other people approve it as is I won't oppose the merge, even if it will have to be changed to display the latest rul first for the real time issue(#88).

rdsaliba commented:
Interesting fact... The default sorting mechanism is not working as expected on this TableView... I have tried changing the object being passed to a Double and Integer but resulted with the same weird sorting... I think, for now it would be best to accept this pull request and attempt to do the sorting in the issue 88

rdsaliba committed:
Merge branch 'develop' into US110_RawData

najimgh reviewed (changes requested):
Tabs are displayed well, the raw data display looks good and the relevant information is present.
However, upon clicking on a specific system to display its SystemInformation page, the following error occurs:
(This error does not affect the raw data display, however)
![Screen Shot 2021-01-22 at 10 55 00 PM](https://user-images.githubusercontent.com/46761237/105568134-1399d380-5d05-11eb-93ce-2a12268e7731.png)

rdsaliba committed:
Display the data with latest first

rdsaliba commented:
Hi @najimgh, I have tested the application from my side and did not get any error. I believe this must be something unrelated to the above changes ...

najimgh reviewed (approved):
After some investigation on the error, it looks like it is unrelated to your PR. Everything else looks good.

PaulMicu commented:
thank you for implementing the ordering from the most recent to the oldest
